waiter/test/waiter/core_test.clj

```diff
@@ -1099,3 +1099,14 @@
 
       (testing "require-authentication"
         (is (= {:source :spnego-handler} (request-handler {:headers {}})))))))
+
+(deftest test-waiter-request?-fn
```

Comment:
The test should also include negative scenarios, e.g. `(is (not (waiter-request?-fn {:headers {"host" "waiter-host-1"}})))`  in the first testing block.

Reply:
Done